pkg/controller/clusterrelocator/clusterrelocator_controller.go

```diff
@@ -0,0 +1,545 @@
+			Name:      relocator.Spec.KubeConfigSecretRef.Name,
+		},
+		kubeConfigSecret,
+	); err != nil {
+		logger.WithError(err).Log(controllerutils.LogLevel(err), "failed to get kubeconfig secret")
+		return false, errors.Wrap(err, "failed to get kubeconfig secret")
+	}
+
+	remoteClientBuilder := r.remoteClusterAPIClientBuilder(kubeConfigSecret)
+	destClient, err := remoteClientBuilder.Build()
+	if err != nil {
+		logger.WithError(err).Warn("could not create a client for the destination cluster")
+		return false, errors.Wrap(err, "could not create a client for the destination cluster")
+	}
+
+	// create namespace
+	switch err := destClient.Create(context.Background(), &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cd.Namespace,
+		},
+	}); {
+	case apierrors.IsAlreadyExists(err):
+		logger.Info("namespace already exists in destination cluster")
+	case err != nil:
+		logger.WithError(err).Log(controllerutils.LogLevel(err), "failed to create namespace in destination cluster")
+		return false, errors.Wrap(err, "failed to create namespace in destination cluster")
+	default:
+		logger.Info("namespace created")
+	}
+
+	cdKey, err := client.ObjectKeyFromObject(cd)
+	if err != nil {
+		logger.WithError(err).Error("could not get object key for clusterdeployment")
+		return false, errors.Wrap(err, "could not get object key for clusterdeployment")
+	}
+	destCD := &hivev1.ClusterDeployment{}
+	switch err := destClient.Get(context.Background(), cdKey, destCD); {
+	case apierrors.IsNotFound(err):
+		logger.Info("clusterdeployment absent in destination cluster")
+	case err != nil:
+		logger.WithError(err).Log(controllerutils.LogLevel(err), "failed to get clusterdeployment in destination cluster")
+		return false, errors.Wrap(err, "failed to get clusterdeployment in destination cluster")
+	default:
+		logger.Info("clusterdeployment already exists in destination cluster")
+		if destCD.Spec.BaseDomain == cd.Spec.BaseDomain {
+			return true, nil
+		}
+		logger.Warn("clusterdeployment in destination cluster does not match the one being relocated")
+		if conds, changed := controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+			cd.Status.Conditions,
+			hivev1.RelocationFailedCondition,
+			corev1.ConditionTrue,
+			"ClusterDeploymentMismatch",
+			"clusterdeployment in destination cluster does not match the one being relocated",
+			controllerutils.UpdateConditionIfReasonOrMessageChange,
+		); changed {
+			cd.Status.Conditions = conds
+			if err := r.Status().Update(context.Background(), cd); err != nil {
+				logger.WithError(err).Log(controllerutils.LogLevel(err), "could not update condition")
+				return false, errors.Wrap(err, "could not update condition")
+			}
+		}
+		return false, nil
+	}
+
+	// move dependent resources
+	for _, t := range typesToMove() {
+		if err := r.moveResources(cd, destClient, t, logger); err != nil {
+			return false, errors.Wrapf(err, "failed to move %T", t)
+		}
```

Comment:
I would like to only copy the secrets and configmaps that are needed by the ClusterDeployment and its dependents. That is a goal I have for a subsequent refinement.

I could pretty easily add code that ignores secrets that have the `kubernetes.io/service-account-token` type. The secrets of `kubernetes.io/dockerconfigjson` type are not as straightforward since that is what is used for the pull secret for the ClusterDeployment.

Reply:
The clusterrelocator controller has been modified to exclude sa token secrets and secret owned by sa token secrets.